[[ Bug 22019 ]] Fix several memory leaks in the field object

runrevmark opened the pull request:
This patch fixes four memory leaks caused by using field features.

The first leak which has been fixed occurred when a field object
had a field-level tabAlign setting. A failure to free the underlying
setting when the field was destroyed caused the leak.

The second leak which has been fixed occurred when fetching the
tabAlign property across multiple paragraphs. A failure to free
intermediate values used to compute whether the setting was mixed
or not caused the leak.

The third leak which has been fixed occurred when importing field
content with htmlText when the htmlText contained tabAlign settings.
A failure to free intermediate values used to compute the final
tabAlign settings caused the leak.

The fourth leak which has been fixed occurred when fetching the
styledText representation of the field or a chunk of the field. An
over-retain of the returned array caused the leak.

livecodeali commented:
I don't think we have any tests that just set / get the tabAlign of a line chunk of a field - might be worth adding them just so that the `PodFieldPropType<MCInterfaceFieldTabAlignments>` stuff is executed

runrevmark committed:
[[ Bug 22019 ]] Fix several memory leaks in the field object

This patch fixes four memory leaks caused by using field features.

The first leak which has been fixed occurred when a field object
had a field-level tabAlign setting. A failure to free the underlying
setting when the field was destroyed caused the leak.

The second leak which has been fixed occurred when fetching the
tabAlign property across multiple paragraphs. A failure to free
intermediate values used to compute whether the setting was mixed
or not caused the leak.

The third leak which has been fixed occurred when importing field
content with htmlText when the htmlText contained tabAlign settings.
A failure to free intermediate values used to compute the final
tabAlign settings caused the leak.

The fourth leak which has been fixed occurred when fetching the
styledText representation of the field or a chunk of the field. An
over-retain of the returned array caused the leak.

runrevmark commented:
@livecodeali I've added a new field tests file for line properties - it exercises all the paragraph/line level properties both when setting on a single line, line ranges and alternating.

livecodeali commented:
@livecode-vulcan review ok bf7abd3

livecode-vulcan commented:
💙 review by @livecodeali ok bf7abd344ba0539ac67a37688dba1bef3676d706

livecode-vulcan commented:
😎 test success bf7abd344ba0539ac67a37688dba1bef3676d706

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/446)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/438)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/438)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/438)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/438)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1960)
* try-community-universal-ios-iphoneos12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos12.1/builds/188)
* try-community-universal-ios-iphonesimulator12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator12.1/builds/187)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1991)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1981)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1980)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2008)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1949)
